src/window_set.c

```diff
@@ -150,6 +166,18 @@ __uint128_t *window_set_get_cookies(window_set_t *set, uint64_t *num_cookies) {
     return res;
 }
 
+void window_set_id_to_cookie(window_set_t *set, const uint8_t *id, __uint128_t *out_cookie) {
+    Word_t *result_hi = j128m_get(&set->id_to_cookie_map_hi, *(__uint128_t *)id);
+    Word_t *result_lo = j128m_get(&set->id_to_cookie_map_lo, *(__uint128_t *)id);
+    if (!result_hi && !result_lo) {
+        *out_cookie = *(__uint128_t *)id;
+        return;
+    }
+
+    ((Word_t *)out_cookie)[1] = *result_hi;
+    ((Word_t *)out_cookie)[0] = *result_lo;
```

Comment:
Technically that breaks strict aliasing rule so I'd do this the bit arithmetic way instead.